cellular-modems-service/src/main.rs

```diff
@@ -0,0 +1,93 @@
+// This file is part of Edgehog.
+//
+// Copyright 2024 SECO Mind Srl
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashMap;
+
+use tracing::{debug, info, level_filters::LevelFilter};
+use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt, EnvFilter};
+use zbus::{
+    dbus_interface,
+    zvariant::{DeserializeDict, SerializeDict, Type},
+    ConnectionBuilder,
+};
+
+pub const SERVICE_NAME: &str = "io.edgehog.CellularModems1";
+
+#[derive(Debug, Default, Clone, DeserializeDict, SerializeDict, Type)]
+#[zvariant(signature = "dict")]
+struct ModemProperties {
+    apn: String,
+    imei: String,
+    imsi: String,
+}
+
+struct CellularModems {
+    modems: HashMap<String, ModemProperties>,
+}
+
+#[dbus_interface(name = "io.edgehog.CellularModems1")]
+impl CellularModems {
+    fn list(&self) -> Vec<String> {
+        self.modems.keys().cloned().collect()
+    }
+
+    fn get(&self, id: String) -> ModemProperties {
+        let modem = self.modems.get(&id);
+
+        if modem.is_none() {
+            debug!("modem {id} not found")
+        }
+
+        modem.cloned().unwrap_or_default()
```

Comment:
you could simplify with
```suggestion
self.modems.get(&id).cloned().unwrap_or_else(|| {
    debug!("modem {id} not found");
    ModemProperties::default()
})
```